### PR TITLE
Add Zabbix configuration recipe

### PR DESCRIPTION
**Pull request by kbushgit**
Signed-off-by: Kostiantyn Bushko <kbushko@intellias.com>
Zabbix configuration recipe allows to configure zabbix_agernt in order to connect to zabbix_server app:

Options:
SOTA_COMM_CONF_ZABBIX_SERVER // set the zabbix "Server" address
SOTA_COMM_CONF_ZABBIX_SERVERACTIVE // zabbix "ServerActive" address
SOTA_COMM_CONF_ZABBIX_HOSTNAME // zabbix "Hostname"

### CODE REVIEW

**Comment by kbushgit** [comment]
> Can you document somewhere the steps you took to get the server running and connected to a client running zabbix?

Yes, I've added comments in OTA-2150, please check an let me know if something is not clear.
In general the server configuration is default and you can use official documentation or links I had provided in the comment.

File: recipes-connectivity/zabbix/zabbix_%.bbappend [hunk 1]
```diff
@@ -0,0 +1,30 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
```
Comment (patrickvacek) [inline recipes-connectivity/zabbix/zabbix_%.bbappend]: What is this line for?

Comment (kbushgit) [reply recipes-connectivity/zabbix/zabbix_%.bbappend]: probably is unnecessary(: thanks!

**Comment by kbushgit** [comment]
I think if we remove the "Hostname" config, this recipe could have some practical sense, even for the production case. 

The Server and ServerActive are ok and they have a practical sense and this option can be applied to the group of the devices, so the only "Hostname"  must have a unique name for each device, and normally this must be set manually by Admin.

So my proposal is to remove "Hostname" option, in this case, it will have a "system hostname" which is ok for the testing purpose.

**Comment by patrickvacek** [comment]
Is the hostname here specific to the device or specific to the server where the information is reported? I had thought it was the server, which would be fine, but if it is per-device, then I agree, we probably shouldn't put it in the image. Is there some way to read it from the actual hostname of the device, or the device ID from aktualizr?

**Comment by kbushgit** [comment]
> Is the hostname here specific to the device or specific to the server where the information is reported? I had thought it was the server, which would be fine, but if it is per-device, then I agree, we probably shouldn't put it in the image. Is there some way to read it from the actual hostname of the device, or the device ID from aktualizr?

Actually the actual hostname will be applied by default, and yes we can obtain actualizr device ID and apply it.

**Comment by patrickvacek** [comment]
> Actually the actual hostname will be applied by default, and yes we can obtain actualizr device ID and apply it.

Cool, that sounds good. IIRC the default hostname on qemu is the hardware ID ("qemux86-64"), which might not be helpful, but if you can change it to the device ID, that'd be great. How does one do that?

**Comment by kbushgit** [comment]
> Cool, that sounds good. IIRC the default hostname on qemu is the hardware ID ("qemux86-64"), which might not be helpful, but if you can change it to the device ID, that'd be great. How does one do that?

For the testing purpose, qemux86-64 is fine, but if someone decides to use Zabbix agent for production, it requires some additional work in run time, I'm not sure it should be a part of aktualizr.
It requires to patch config file and restart zabbix service. 
So we can add the script, as a part of Zabbix recipe, which will be able to run aktualizr-info grep an ID and patch Zabbix config file and then restart zabbix service, this script can be run as a system-d unit with the type OneShot.

**Comment by patrickvacek** [comment]
> For the testing purpose, qemux86-64 is fine, but if someone decides to use Zabbix agent for production, it requires some additional work in run time, I'm not sure it should be a part of aktualizr.

Agreed. No need to worry about it for now, then.

> It requires to patch config file and restart zabbix service.
> So we can add the script, as a part of Zabbix recipe, which will be able to run aktualizr-info grep an ID and patch Zabbix config file and then restart zabbix service, this script can be run as a system-d unit with the type OneShot.

Thanks for the details! This can probably also go in the document that describes how to set all of this up.

**Pull request merged**